### ISSUE TITLE
Question about Learning rate

### ISSUE THREAD

**Issue by Colanim** [nlpyang/BertSum#53]
According to [your paper](https://arxiv.org/pdf/1903.10318.pdf) :

![image](https://user-images.githubusercontent.com/43774355/60239437-ac2d5f00-98e8-11e9-888a-ff5c784d35cb.png)

As you said, it follows [Attention is all you need](https://arxiv.org/pdf/1706.03762.pdf) :

![image](https://user-images.githubusercontent.com/43774355/60239482-dd0d9400-98e8-11e9-9b45-1828f961d796.png)

But in your case, **what is the reason you choose `2e-3` as initial learning rate ?**

If we follow the formula of Vaswani, we have :

d_model = 768 (because we use BERT-base)
d_model^-0.5 ~= 0.036

So where this `2e-3` comes from ?

---

**Also, why choosing 10 000 warmup steps ?** 

Original Transformer paper used 4 000 warmup steps - 100 000 total steps.

BERT paper used 10 000 warmup steps - 1 000 000 total steps, but they didn't use noam decay method, just linear decay.

**Comment by nlpyang** [comment]
All these hyperparameters are first set by experience and then tested by experiments as in all neural models. The learning rate for finetuning BERT should be smaller than traiking a model from scratch as in the original transformer paper. 

You can alao use your own learning schedule.

**Comment by Colanim** [comment]
Thanks for you answer 👍
